### ISSUE TITLE
Node with the name XXXXXXX already exists under parentId YYYYYYY

### ISSUE THREAD

**Issue by amaxcz** [yadayada/acd_cli#152]
Uploading 400G music...
acdcli --utf upload --max-connections 8 --max-retries 10 --overwrite /Volumes/Data/Music /

sometime its crashes with errors like this:
```
-- Logging error ---   ]  39.3% of  429GiB   6750/19242    2.3MB/s   1d 10h
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 409, {"logref":"df8ebbe3-62ee-11e5-89e3-6524f3d353c9","message":"Node with the name 4-01 dakar.m4a already exists under parentId ejEsecv3T2CzmVIAgDO5sg conflicting NodeId: m-hDjmIITciXWZ9fNSzbpQ","code":"NAME_ALREADY_EXISTS","info":{"nodeId":"m-hDjmIITciXWZ9fNSzbpQ"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 333, in upload_file
    'If you want to overwrite, please sync and try again.' % short_nm)
Message: 'Uploading "4-01 Dakar.m4a" failed. Name collision with non-cached file. If you want to overwrite, please sync and try again.'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 409, {"logref":"dd6e261b-62ee-11e5-be3b-f912e8a3808b","message":"Node with the name 13 - philadelphia freedom.m4a already exists under parentId GEgt6s5zQv-9Dob4N1YnIQ conflicting NodeId: ZF-nEjXHRGSAHMR9SYjQMw","code":"NAME_ALREADY_EXISTS","info":{"nodeId":"ZF-nEjXHRGSAHMR9SYjQMw"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 333, in upload_file
    'If you want to overwrite, please sync and try again.' % short_nm)
Message: 'Uploading "13 - Philadelphia Freedom.m4a" failed. Name collision with non-cached file. If you want to overwrite, please sync and try again.'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 500, {"logref":"aa221a4d-62f1-11e5-bff1-eb6804221a30","message":"Internal failure","code":""}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 347, in upload_file
    'Uploading "%s" failed. Code: %s, msg: %s' % (short_nm, e.status_code, e.msg))
Message: 'Uploading "04 Fat Old Sun.m4a" failed. Code: 500, msg: {"logref":"aa221a4d-62f1-11e5-bff1-eb6804221a30","message":"Internal failure","code":""}'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 400, {"message":"Cannot complete upload"}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 347, in upload_file
    'Uploading "%s" failed. Code: %s, msg: %s' % (short_nm, e.status_code, e.msg))
Message: 'Uploading "14 Damage Case.m4a" failed. Code: 400, msg: {"message":"Cannot complete upload"}'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 400, {"message":"Cannot complete upload"}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 347, in upload_file
    'Uploading "%s" failed. Code: %s, msg: %s' % (short_nm, e.status_code, e.msg))
Message: 'Uploading "01 The Unnamed Feeling.m4a" failed. Code: 400, msg: {"message":"Cannot complete upload"}'
Arguments: ()
beruska:~ amaxcz$ acdcli --utf upload --max-connections 8 --max-retries 10 --overwrite /Volumes/Data/Music /
^Cberuska:~ amaxcz$ acdcli --utf upload --max-connections 8 --max-retries 30 --overwrite /Volumes/Data/Music /
--- Logging error ---     ]  41.4% of  429GiB   6914/19242    2.3MB/s   1d 08h
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 409, {"logref":"e331c33f-630a-11e5-8b25-05a4f41ed579","message":"Node with the name 13 - philadelphia freedom.m4a already exists under parentId GEgt6s5zQv-9Dob4N1YnIQ conflicting NodeId: ZF-nEjXHRGSAHMR9SYjQMw","code":"NAME_ALREADY_EXISTS","info":{"nodeId":"ZF-nEjXHRGSAHMR9SYjQMw"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 333, in upload_file
    'If you want to overwrite, please sync and try again.' % short_nm)
Message: 'Uploading "13 - Philadelphia Freedom.m4a" failed. Name collision with non-cached file. If you want to overwrite, please sync and try again.'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 409, {"logref":"e5222145-630a-11e5-9c2f-27ffbce2b4e9","message":"Node with the name 4-01 dakar.m4a already exists under parentId ejEsecv3T2CzmVIAgDO5sg conflicting NodeId: m-hDjmIITciXWZ9fNSzbpQ","code":"NAME_ALREADY_EXISTS","info":{"nodeId":"m-hDjmIITciXWZ9fNSzbpQ"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 333, in upload_file
    'If you want to overwrite, please sync and try again.' % short_nm)
Message: 'Uploading "4-01 Dakar.m4a" failed. Name collision with non-cached file. If you want to overwrite, please sync and try again.'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 161, in upload_file
    raise RequestError(r.status_code, r.text)
acdcli.api.common.RequestError: RequestError: 409, {"logref":"e99a1500-630a-11e5-bc57-5371a6d7996f","message":"Node with the name 01 broken, beat \u0026 scarred.m4a already exists under parentId w0JSISbkSySNCLb9umxO0Q conflicting NodeId: U5D9cCRMR82FltY0csyqGA","code":"NAME_ALREADY_EXISTS","info":{"nodeId":"U5D9cCRMR82FltY0csyqGA"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 333, in upload_file
    'If you want to overwrite, please sync and try again.' % short_nm)
Message: 'Uploading "01 Broken, Beat & Scarred.m4a" failed. Name collision with non-cached file. If you want to overwrite, please sync and try again.'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 349, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1088, in request
    self._send_request(method, url, body, headers)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1126, in _send_request
    self.endheaders(body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1084, in endheaders
    self._send_output(message_body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 926, in _send_output
    self.send(message_body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 885, in send
    self.sock.sendall(datablock)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/ssl.py", line 723, in sendall
    v = self.send(data[count:])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/ssl.py", line 684, in send
    v = self._sslobj.write(data)
BrokenPipeError: [Errno 32] Broken pipe

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 597, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/util/retry.py", line 245, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/packages/six.py", line 309, in reraise
    raise value.with_traceback(tb)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 349, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1088, in request
    self._send_request(method, url, body, headers)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1126, in _send_request
    self.endheaders(body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 1084, in endheaders
    self._send_output(message_body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 926, in _send_output
    self.send(message_body)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/http/client.py", line 885, in send
    self.sock.sendall(datablock)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/ssl.py", line 723, in sendall
    v = self.send(data[count:])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/ssl.py", line 684, in send
    v = self._sslobj.write(data)
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/common.py", line 46, in decorated
    return func(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/backoff_req.py", line 90, in _request
    r = requests.request(type_, url, headers=headers, timeout=timeout, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/api.py", line 50, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 328, in upload_file
    deduplication=dedup)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/content.py", line 158, in upload_file
    headers={'Content-Type': m.content_type})
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/backoff_req.py", line 104, in post
    return self._request('POST', url, acc_codes, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/api/common.py", line 48, in decorated
    raise RequestError(RequestError.CODE.CONN_EXCEPTION, e.__str__())
acdcli.api.common.RequestError: RequestError: 1000, ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
ValueError: underlying buffer has been detached
Call stack:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/acdcli/utils/threading.py", line 40, in _worker_task
    rr = f()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 191, in wrapped
    ret_val = f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/acd_cli.py", line 347, in upload_file
    'Uploading "%s" failed. Code: %s, msg: %s' % (short_nm, e.status_code, e.msg))
Message: 'Uploading "2-04 Billy Preston.m4a" failed. Code: 1000, msg: (\'Connection aborted.\', BrokenPipeError(32, \'Broken pipe\'))'
Arguments: ()
```

**Comment by yadayada** [comment]
Your node cache is outdated. You need to wait a minute or so, sync and try again. This should avoid code 409 errors.

Code 500 errors may be anything, unfortunately.

Omit the utf flag or use the latest dev commit to avoid the logging errors.

**Comment by amaxcz** [comment]
dont close it!!!!! again... I has never touched anything. So.. I'm start some command. and going to sleep. Returns and watch an error! So. its the internal problem of the multithreading upload. LOOK.

**Comment by yadayada** [comment]
I would need a --verbose or --debug log to verify that e.g. "4-01 Dakar.m4a" wasn't uploaded before.

This may be a duplicate of #100.

**Comment by amaxcz** [comment]
so.. again.. its was synced inititally. synced before starting upload job. only 1 client. multithreading upload. I havent this folder (Music) at the remote side at all. so. I synced. put a upload job to console and go to the bed. I have returned. Job died. synced again. starts it again. Working sometime.. and die again on the different file but same error. its a but with multithreading upload. somewhere thread do a job, and second thread going to the collision state.

**Comment by amaxcz** [comment]
you should not broke/abort all remained threads, the uploading process should not die if this error occurs.  its a variant of unsynced thread job (bug somewhere)

**Comment by yadayada** [comment]
>somewhere thread do a job, and second thread going to the collision state.

And you believe this because of what?

> you should not broke/abort all remained threads

Usually, transfer job threads don't die. In this case they do because the logging is/was broken because of the utf bug (#152).

>Omit the utf flag or use the latest dev commit to avoid the logging errors.

Do this before you reply again.

**Comment by amaxcz** [comment]
ok aggreed. thank to you.
will try :)
